cdap-ui/app/cdap/components/Reports/ReportsList/index.js

```diff
@@ -16,43 +16,47 @@
 
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import {MyReportsApi} from 'api/reports';
 import Customizer from 'components/Reports/Customizer';
 import {humanReadableDate} from 'services/helpers';
 import IconSVG from 'components/IconSVG';
-import ReportsStore, {ReportsActions} from 'components/Reports/store/ReportsStore';
 import {connect} from 'react-redux';
 import Duration from 'components/Duration';
 import { Link } from 'react-router-dom';
 import {getCurrentNamespace} from 'services/NamespaceStore';
+import {listReports} from 'components/Reports/store/ActionCreator';
+import {Observable} from 'rxjs/Observable';
+import classnames from 'classnames';
+import ActionPopover from 'components/Reports/ReportsList/ActionPopover';
 
 require('./ReportsList.scss');
 
 class ReportsListView extends Component {
   static propTypes = {
-    reports: PropTypes.array
+    reports: PropTypes.array,
+    activeId: PropTypes.string
   };
 
   componentWillMount() {
-    let params = {
-      offset: 0,
-      limit: 20
-    };
-
-    MyReportsApi.list(params)
-      .subscribe((res) => {
-        ReportsStore.dispatch({
-          type: ReportsActions.setList,
-          payload: res
-        });
-      });
+    listReports();
+    this.interval$ = Observable.interval(10000)
```

Comment:
I understand that we can't do polling on action creators. If we do a `POLL` type request here, 
- the request goes out from client only once
- the response comes back only when there is a change in the data from the node proxy.

I would vote for moving the action creator listreports polling here and send out a `STOP-POLL` when the component is unmounting. 

Event though the exisitng behavior sends out a websocket message once every 10 seconds, it still updates the store on each response, whether there is a change in the data or not. We could avoid this which could avoid repeated re-render in UI if we make a `POLL` request from within the component.

Reply:
I agree, but actually this list polling will need to go away once I do pagination, because then I have to change the offset of the request. The poll needs to only happen on the rows that are still generating, I don't need to keep polling the entire list.